### DISCUSSION TITLE
Set Exclusive constrain on two properties

### DISCUSSION THREAD

**Discussion by clmnin** [geldata/gel#1974] [Q&A] [answered]
How do I make a group of properties exclusive?

Here is an example of this in postgres
```sql
CREATE TABLE someTable (
    id serial primary key,
    col1 int NOT NULL,
    col2 int NOT NULL,
    unique (col1, col2)
)
```

I did read the documentation for [std::exclusive](https://www.edgedb.com/docs/datamodel/constraints#constraint::std::exclusive). According to the doc I can have a unique constrain if using `links`. So in that case I would have to use

```EdgeQL
type ExclusiveColumns {
    property col1 -> int64;
    property col2 -> int64;
}

type SomeTable {
    link columns -> ExclusiveColumns {
        constraint exclusive;
    }
}
```

This approach is actually dependent on the UUID of the object created in `INSERT ExclusiveColumns` and so I can't create SomeTable in one go like
```
INSERT SomeTable {
    columns := (
        INSERT ExclusiveColumns {
            col1 := 1,
            col2 := 2
        }
    )
}
```
And so I'll have to make two requests to the EdgeDB server to achieve this. But is this the only way?

**Comment by elprans** [comment, marked as answer]
Here's how:

```
type ExclusiveColumns {
    property col1 -> int64;
    property col2 -> int64;

    constraint exclusive on ( (.col1, .col2) );
}
```

@vpetrovykh, please add an example to `std::exclusive` reference.
